### ISSUE TITLE
[Bug] Code validation (syntax error) does not work for php language

### ISSUE THREAD

**Issue by SteelManITA** [microsoft/monaco-editor#4781]
### Reproducible in vscode.dev or in VS Code Desktop?

- [x] Not reproducible in [vscode.dev](https://vscode.dev) or VS Code Desktop

### Reproducible in the monaco editor playground?

- [ ] Not reproducible in [the monaco editor playground](https://microsoft.github.io/monaco-editor/playground.html)

### Monaco Editor Playground Code

```typescript
<?php
// The next line contains a syntax error:
if () {
  return "The parser recovers from this type of syntax error"
}
?>
```

### Reproduction Steps

Write a syntax error, like missing condition in if statement or missing a ;

### Actual (Problematic) Behavior

Code validation (syntax error) does not work for php language

It does not work in demo either (microsoft.github.io/monaco-editor)

If you select PHP language, the first comment is "the next line contains a syntax error". But the "next line" (which has a syntax error) is not highlighted

### Expected Behavior

Code validation works

**Comment by rcjsuen** [comment]
@SteelManITA Monaco does not support validating PHP syntax.

**Comment by SteelManITA** [comment]
@rcjsuen I don't understand... Why then on the page https://microsoft.github.io/monaco-editor is there actually an example in PHP that refers to validation?

In any case, is there currently no way to have even basic validation of PHP syntax?

**Comment by rcjsuen** [comment]
> [@rcjsuen](https://github.com/rcjsuen) I don't understand... Why then on the page https://microsoft.github.io/monaco-editor is there actually an example in PHP that refers to validation?

Which example are you referring to?

> In any case, is there currently no way to have even basic validation of PHP syntax?

I doubt it personally but I don't work for Microsoft. You'd probably need to hook it up with a language server.

**Comment by SteelManITA** [comment]
@rcjsuen 

![Image](https://github.com/user-attachments/assets/5fbcb741-211a-4fb3-8192-f23b4d2adba6)

**Comment by rcjsuen** [comment]
> [@rcjsuen](https://github.com/rcjsuen) I refer to the example that appears when selecting php on the homepage (screnshot below)

I agree it's kind of a strange design choice but unfortunately the reality is that it does not have PHP syntax validation out-of-the-box.

**Comment by SteelManITA** [comment]
Ok, thanks for clarifying. Do you know any packages I can use to achieve my
goal?

Il mar 17 dic 2024, 12:34 Remy Suen ***@***.***> ha scritto:

**Comment by rcjsuen** [comment]
> Ok, thanks for clarifying. Do you know any packages I can use to achieve my goal?

You should read up on the [language server protocol](https://microsoft.github.io/language-server-protocol/) and then look at [TypeFox/monaco-languageclient](https://github.com/TypeFox/monaco-languageclient/). They have examples that integrate with Java and Python language servers that you can reference.
